formats/protobuf/src/main/java/io/cloudevents/protobuf/ProtoDataWrapper.java

```diff
@@ -53,17 +60,22 @@ public boolean equals(Object obj) {
         // Now compare the actual data
         ProtoDataWrapper rhs = (ProtoDataWrapper) obj;
 
-        if (this.getMessage() == rhs.getMessage()){
-            return true;
-        }
+        final Any lhsAny = getAny();
+        final Any rhsAny = rhs.getAny();
 
         // This is split out for readability.
-        //  Compare the content in terms onf an 'Any'.
-        //  - Verify the types match
-        //  - Verify the values match.
+        //  1. Sanity compare the 'Any' references.
+        //  2. Compare the content in terms onf an 'Any'.
+        //     - Verify the types match
+        //     - Verify the values match.
 
-        final Any lhsAny = getAsAny(this.getMessage());
-        final Any rhsAny = getAsAny(rhs.getMessage());
+        if (lhsAny == rhsAny) {
+            return true;
+        }
+
+        if ((lhsAny == null) || (rhsAny == null)) {
+            return false;
+        }
```

Comment:
lhsAny or rhsAny cannot be null since the constructor has:
```
        Objects.requireNonNull(protoMessage);

        if (protoMessage instanceof Any) {
            protoAny = (Any) protoMessage;
        } else {
            protoAny = Any.pack(protoMessage);
        }
```

```suggestion
```

Reply:
;-) ... thanks